src/expire.c

```diff
@@ -440,7 +440,13 @@ void expireGenericCommand(client *c, long long basetime, int unit) {
         addReply(c, shared.cone);
         return;
     } else {
+        robj *expire_when, *pexpireat;
         setExpire(c,c->db,key,when);
+        pexpireat = createStringObject("PEXPIREAT", 9);
```

Comment:
Hi @kingpeterpaule , for "PEXPIREAT", i think using a preallocated shared object is a better idea. Redis uses this kind of optimization for many times.

Reply:
Yes, I agree.